<h1>Welcome to my GitHub page! <img src="https://emojis.slackmojis.com/emojis/images/1531849430/4246/blob-sunglasses.gif?1531849430" width="30"/></h1>

<p>I'm Silvia, a PhD candidate at the <b>Autonomous University of Madrid (UAM), Spain</b> 
<img src="https://img.icons8.com/color/48/000000/spain-circular.png" width="13"/>.  
I’m currently finishing my doctoral thesis in <b>computational and theoretical microbial ecology</b> at 
<a href="https://www.me-genomics.com/">Aguirre de Cárcer's Lab</a>.</p>

<h3>🔭 My work focuses on...</h3>

- Developing and maintaining <b>bioinformatic workflows</b> for microbial community analysis 🧬  
- Integrating tools for <b>functional annotation</b> (e.g. eggNOG-mapper, FunctionInk) 🔎  
- <b>Metabolic modeling</b> using FBA-based tools (ReFramed, CarveMe, Smetana, MISO) ⚙️  
- Designing <b>reproducible pipelines</b> and simulation tools (dilgrowth package) 🧪  

<h3>💻 Some programming languages and tools I have worked with</h3>
<p>
<img alt="Python" src="https://img.shields.io/badge/-Python-3776AB?style=flat-square&logo=python&logoColor=white" />
<img alt="R" src="https://img.shields.io/badge/-R-276DC3?style=flat-square&logo=R&logoColor=white" />
<img alt="bash" src="https://img.shields.io/badge/-Bash-1f2329?style=flat-square&logo=gnu-bash&logoColor=white" />
<img alt="Linux" src="https://img.shields.io/badge/-Linux-FCC624?style=flat-square&logo=linux&logoColor=black" />
<img alt="C++" src="https://img.shields.io/badge/-C++-00599C?style=flat-square&logo=cplusplus&logoColor=white" />
<img alt="Git" src="https://img.shields.io/badge/-Git-F05032?style=flat-square&logo=git&logoColor=white" />
<img alt="Linux" src="https://img.shields.io/badge/-Linux-FCC624?style=flat-square&logo=linux&logoColor=black" />
<img alt="Perl" src="https://img.shields.io/badge/-Perl-00bcde?style=flat-square&logo=Perl&logoColor=white" />
<img alt="MATLAB" src="https://img.shields.io/badge/-MATLAB-ea6e00?style=flat-square&logo=M&logoColor=white" />
</p>

<h3>⚙️ I have experience with...</h3>

- Studying microbial community assembly through identification of phylogenetic core groups, functional annotation, and metabolic modeling (FBA).
- Managing HPC environments (CBM, CCC-UAM clusters) and workflow optimization using SLURM.
- Programming for <a href="https://github.com/Opentrons/opentrons">automated lab platforms (Opentrons OT-2)</a>.
- Collaborating with national and international research teams (Spain 🇪🇸, Norway 🇳🇴, Japan 🇯🇵).
- Supporting researchers in metagenomic and metatranscriptomic data processing.

<h3>👯 I’m looking to collaborate on...</h3>
- Applying <b>bioinformatics and computational tools</b> to advance scientific research 🧬<br>
- Development and integration of open-source bioinformatic tools 🔧

<h3>📫 How to reach me</h3>

- 📧 <a href="mailto:stalaveramarcos@protonmail.com">stalaveramarcos@protonmail.com</a>  
- 📍Faculty of Science - Biology Building, C/ Darwin 2, 28049 Madrid 
- 🗣️ You can talk to me in <b>Spanish, English, French, or Japanese</b>, and I’m learning <b>Norwegian and Chinese</b>.

---
## 💻 Featured Projects and Repositories

<br>

<img align="left" alt="R logo" src="https://img.shields.io/badge/-R-276DC3?style=flat-square&logo=R&logoColor=white" width="30"/>
&nbsp;&nbsp;&nbsp;
<b><a href="https://github.com/silvtal/dilgrowth">dilgrowth</a></b>
<br>
This package includes command line scripts and standalone R/c++ functions for simulating growth-dilution experiments. These simulations include neutral drift and optionally functional groups with different group fitness. 
<br clear="left"/>
<br>

<img align="left" alt="R logo" src="https://img.shields.io/badge/-R-276DC3?style=flat-square&logo=R&logoColor=white" width="30"/>
&nbsp;&nbsp;&nbsp;
<b><a href="https://github.com/silvtal/correlation_graphs">correlation_graphs</a></b>
<br>
Based on Goyal et al.'s (2018) method for inferring statistically significant correlations between OTUs.
<br clear="left"/>
<br>

<img align="left" alt="Python logo" src="https://img.shields.io/badge/-Python-3776AB?style=flat-square&logo=python&logoColor=white" width="30"/>
&nbsp;&nbsp;&nbsp;
<b><a href="https://github.com/silvtal/SBML_to_graph">SBML_to_graph</a></b>
<br>
A few rudimentary functions to generate metabolite graphs from SBML models. Uses ReFramed and one CarveMe function.
<br clear="left"/>
<br>

<img align="left" alt="Python logo" src="https://img.shields.io/badge/-Python-3776AB?style=flat-square&logo=python&logoColor=white" width="30"/>
&nbsp;&nbsp;&nbsp;
<b><a href="https://github.com/silvtal/custom_ot2_utils">custom_ot2_utils</a></b>
<br>
Opentrons OT2 robot functions or custom labware definitions that we feel other people might find useful
<br clear="left"/>
<br>

<img align="left" alt="HTML logo" src="https://img.shields.io/badge/-HTML5-E34F26?style=flat-square&logo=html5&logoColor=white" width="30"/>
&nbsp;&nbsp;&nbsp;
<b><a href="https://github.com/silvtal/TFM">TFM</a></b>
<br>
Este repositorio recoge los scripts desarrollados para automatizar los análisis correspondientes al Trabajo de Fin de Máster "Modelización metabólica de comunidades microbianas estables crecidas con fuentes de carbono y energía únicas y simples" (Máster en Bioinformática y Biología Computacional, Universidad Autónoma de Madrid)
<br clear="left"/>

<img align="left" alt="JavaScript logo" src="https://img.shields.io/badge/-JavaScript-F7DF1E?style=flat-square&logo=javascript&logoColor=black" width="30"/>
&nbsp;&nbsp;&nbsp;
<b><a href="https://github.com/silvtal/wordle-jp">wordle-jp</a></b>
<br>
Japanese adaptation of Josh Wardle's Wordle
<br clear="left"/>
<br>


---

Last update: October 2025<br>
